hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/operators/common/HazelcastAggFunction.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.validate.operators.common;
+
+import com.hazelcast.sql.impl.calcite.validate.HazelcastCallBinding;
+import org.apache.calcite.sql.SqlAggFunction;
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlOperandCountRange;
+import org.apache.calcite.sql.type.SqlOperandCountRanges;
+import org.apache.calcite.sql.type.SqlOperandTypeChecker;
+import org.apache.calcite.sql.type.SqlOperandTypeInference;
+import org.apache.calcite.sql.type.SqlReturnTypeInference;
+import org.apache.calcite.util.Optionality;
+
+import static com.hazelcast.sql.impl.calcite.validate.operators.HazelcastReturnTypeInference.wrap;
+
+/**
+ * A common subclass for aggregate functions (in Jet).
+ * <p>
+ * See {@link HazelcastOperandTypeCheckerAware} for motivation.
+ */
+public abstract class HazelcastAggFunction extends SqlAggFunction implements HazelcastOperandTypeCheckerAware {
+
+    public HazelcastAggFunction(
+            String name,
+            SqlKind kind,
+            SqlReturnTypeInference returnTypeInference,
+            SqlOperandTypeInference operandTypeInference,
+            SqlOperandTypeChecker operandTypeChecker,
+            SqlFunctionCategory funcType,
+            boolean requiresOrder,
+            boolean requiresOver,
+            Optionality requiresGroupOrder
+    ) {
+        super(name, null, kind, wrap(returnTypeInference), operandTypeInference, operandTypeChecker,
+                funcType, requiresOrder, requiresOver, requiresGroupOrder);
+    }
+
+    @Override
+    public final boolean checkOperandTypes(SqlCallBinding callBinding, boolean throwOnFailure) {
+        HazelcastCallBinding bindingOverride = prepareBinding(callBinding);
+
+        return checkOperandTypes(bindingOverride, throwOnFailure);
+    }
+
+    @Override
+    public SqlOperandCountRange getOperandCountRange() {
+        return SqlOperandCountRanges.between(1, 1);
```

Comment:
Is there a reason to limit all aggregate operators to a single operand only? I think this should be decided on a per-operator basis (e.g. see `LISTAGG` function).

Reply:
All current agg functions have 1 argument. If some future function will have more, it can override.

Reply:
Some Calcite functions support more than one argument. My question is why do we restrict it for all operators, instead of setting the proper operand counts on a case-by-case basis? If we know for sure, that some functions require more than one argument, then we may implement the base class properly right from the beginning, especially given that this is a small change, the requires only slight code reorganization (e.g. move the method to concrete classes, or pass the argument count from the constructor, etc.).

Reply:
We're not _restricting_ it. 100% of functions currently have 1 argument, if we add `LISTAGG` one day, then 90% of the functions will have 1 argument. Think of it as a good default for most cases, overriding the method is a third option of providing this information from the subclass.